pages/certificates/_components/DNIForm.js

```diff
@@ -39,18 +40,13 @@ export default function DNIForm({ state = null, setState, dispatch, dni }) {
     setIsLoading(true);
     setTimeout(async () => {
       setIsLoading(false);
-      if (dni !== DNIValue) {
+      const res = await getFetch(`/users/${DNI}/credentials`);
+      if (res.data.errors) {
         setState("notFound");
-      } else {
-        const { error, payload } = await dispatch(getCredentials());
-        if (!payload.data.errors) {
-          router.push("/user");
-        } else {
-          setState("notFound");
-        }
+      } else if (res.data.user) {
+        router.push("/user");
       }
       setEdit(true);
-      setState("success");
     }, 1000);
```

Comment:
timers are not a good idea to be used in an async request, async methods are unpredictable.